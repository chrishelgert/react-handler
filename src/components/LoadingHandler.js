import React, { PropTypes } from 'react';
import { LoadingMessage } from './Messages';

/**
 * This Mini-Handler handles only the loading-state
 * or display the given Component.
 */
const LoadingHandler = ({
  LoadingComponent,
  Component,
  componentProps = {},
  loading = true,
}) => {
  if (loading) {
    return LoadingComponent ? <LoadingComponent /> : <LoadingMessage />;
  }

  return <Component {...componentProps} />;
};

LoadingHandler.propTypes = {
  LoadingComponent: PropTypes.func,
  Component: PropTypes.func.isRequired,
  componentProps: PropTypes.object,
  loading: PropTypes.bool,
};

export default LoadingHandler;
