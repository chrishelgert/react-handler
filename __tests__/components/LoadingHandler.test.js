import React from 'react'
import renderer from 'react-test-renderer'
import LoadingHandler from '../../src/components/LoadingHandler'

const TestComponent = () => <h1>Test</h1>
const LoadingComponent = () => <img src='/img/spinner.svg' alt='loading...' />

describe('LoadingHandler', () => {
  test('returns the component when called with loading false', () => {
    const tree = renderer.create(
      <LoadingHandler loading={false}>
        <TestComponent />
      </LoadingHandler>
    )
    expect(tree.toJSON()).toMatchSnapshot()
  })

  test('returns null when called with loading false and no component', () => {
    const tree = renderer.create(
      <LoadingHandler loading={false} />
    )
    expect(tree).toMatchSnapshot()
  })

  test('returns the LoadingComponent', () => {
    const tree = renderer.create(
      <LoadingHandler LoadingComponent={LoadingComponent}>
        <TestComponent />
      </LoadingHandler>
    )
    expect(tree.toJSON()).toMatchSnapshot()
  })

  test('returns the LoadingMessage with given loading message', () => {
    const tree = renderer.create(
      <LoadingHandler message='keep keep loading'>
        <TestComponent />
      </LoadingHandler>
    )
    expect(tree.toJSON()).toMatchSnapshot()
  })

  test('returns the LoadingMessage when called without LoadingComponent', () => {
    const tree = renderer.create(
      <LoadingHandler>
        <TestComponent />
      </LoadingHandler>
    )
    expect(tree.toJSON()).toMatchSnapshot()
  })
})
