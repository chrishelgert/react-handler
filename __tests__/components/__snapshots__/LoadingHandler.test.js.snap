// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingHandler returns null when called with loading false and no component 1`] = `null`;

exports[`LoadingHandler returns the LoadingComponent 1`] = `
<img
  alt="loading..."
  src="/img/spinner.svg"
/>
`;

exports[`LoadingHandler returns the LoadingMessage when called without LoadingComponent 1`] = `
<div
  className="handler--loading"
>
  loading...
</div>
`;

exports[`LoadingHandler returns the LoadingMessage with given loading message 1`] = `
<div
  className="handler--loading"
>
  keep keep loading
</div>
`;

exports[`LoadingHandler returns the component when called with loading false 1`] = `
<h1>
  Test
</h1>
`;
